packages/components/src/components/common/inputtag/inputBadge.tsx

```diff
@@ -0,0 +1,79 @@
+import { forwardRef, type ElementRef } from 'react';
+import { useField } from '../field/field';
+import { IvyIcon } from '../icon/icon';
+import { inputBadge, inputBadgeIcon, inputBadgeLine, inputBadgeOutput } from './inputBadge.css';
+
+import type { IvyIcons } from '@axonivy/ui-icons';
+
+export type InputBadgeProps = {
+  value: string;
+  badgeRegex: BadgeRegex[];
+};
+
+type BadgeRegex = { delimiter: { start: string; end: string }; icon: IvyIcons; badgeTextGen: (text: string) => string };
+
+export type BadgeTextProps = { icon: IvyIcons | undefined; text: string };
+
+const Badge = ({ badge }: { badge: BadgeTextProps }) => {
+  if (badge.icon) {
+    return (
+      <span className={inputBadge}>
+        <span className={inputBadgeIcon}>
+          <IvyIcon icon={badge.icon} />
+        </span>
+        {badge.text}
+      </span>
+    );
+  }
+  return <span>{badge.text}</span>;
```

Comment:
I think a text without an icon should not be a badge. So the badge component should only be used for badges.
And I'm also not sure why you need to wrap the icon inside a span...
```suggestion
  (   
    <Flex gap={1} alignItems='center' className={inputBadge}>
      {icon && <IvyIcon icon={badge.icon} />}
      {text}
    </Flex>
  );
```